src/metabase/lib/query.cljc

```diff
@@ -171,14 +202,14 @@
               (map (fn [[stage-number stage]]
                      (lib.util.match/replace stage
                        [:field
-                        (opts :guard (every-pred map? (complement (some-fn :base-type :effective-type))))
+                        (opts :guard (every-pred map? (complement (every-pred :base-type :effective-type))))
```

Comment:
I believe this `(map... )`should be calling `add-effective-type-to-coereced-fields` and merging the work between the two (expression types and `base-type` from here, move to the fn)

Reply:
Now `add-types-to-fields` is called here and expression types are added to its result.